modules/cssn/src/Plugin/Util/ProjectLookup.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Drupal\cssn\Plugin\Util;
+
+use Drupal\webform\Entity\WebformSubmission;
+
+/**
+ * Lookup connected Project+ nodes.
+ *
+ * @ProjectLookup(
+ *   id = "project_lookup",
+ *   title = @Translation("Project Lookup"),
+ *   description = @Translation("Lookup Users project+ connections for community
+ *   persona.")
+ * )
+ */
+class ProjectLookup {
+  /**
+   * Store project submissions.
+   * $var array
+   */
+  private $projects;
+
+  /**
+   * Array of sorted projects.
+   * $var array
+   */
+  private $projects_sorted;
+
+  /**
+   * Function to return projects.
+   */
+  public function __construct($project_fields, $project_user_id, $project_user_email, $public = FALSE) {
+    $this->runQuery($project_fields, $project_user_id, $project_user_email);
+  }
+
+  /**
+   * Function to Run entity query by type.
+   */
+  public function runQuery($project_fields, $project_user_id, $project_user_email) {
+    $query = \Drupal::database()->select('webform_submission_data', 'wsd');
+    $orGroup = $query->orConditionGroup()
+      ->condition('wsd.value', $project_user_id)
+      ->condition('wsd.value', $project_user_email);
+    $query->fields('wsd', ['sid', 'name']);
+    $query->condition($orGroup);
+    $query->condition('wsd.webform_id', 'project');
+    $result = $query->execute()->fetchAll();
+
+    $query_flag = \Drupal::database()->select('flagging', 'f');
+    $query_flag->fields('f', ['entity_id', 'flag_id']);
+    $query_flag->condition('f.uid', $project_user_id);
+    $query_flag->condition('f.flag_id', 'interested_in_project');
+    $result_flag = $query_flag->execute()->fetchAll();
+    $flagged_results = array_map(function($result_flag) {
+      return (object) [
+        'sid' => $result_flag->entity_id,
+        'name' => 'interested_in_project',
+      ];
+    }, $result_flag);
+    $result = array_merge($result, $flagged_results);
+    if ($result != NULL) {
+      foreach ($result as $project_result) {
+        $wf = WebformSubmission::load($project_result->sid);
+        $wf_lookup = $wf->getData();
+        $this->projects[] = [
+          'title' => $wf_lookup['project_title'],
+          'name' => $project_fields[$project_result->name],
```

Comment:
Perhaps the issue in line 125 could be related to this warning? Not sure why $project_result->name would be "mentors" or "tags"?

```Warning: Undefined array key "mentors" in Drupal\cssn\Plugin\Util\ProjectLookup->runQuery() (line 68 of modules/custom/access/modules/cssn/src/Plugin/Util/ProjectLookup.php).
Warning: Undefined array key "tags" in Drupal\cssn\Plugin\Util\ProjectLookup->runQuery() (line 68 of modules/custom/access/modules/cssn/src/Plugin/Util/ProjectLookup.php).```